pkg/cmd/run.go

```diff
@@ -411,7 +411,15 @@ func fetchSecrets(localConfig models.ScopedOptions, enableCache bool, fallbackOp
 
 	statusCode, respHeaders, response, httpErr := http.DownloadSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value, format, nameTransformer, etag, dynamicSecretsTTL, secretNames)
 	if !httpErr.IsNil() {
-		if fallbackOpts.enable {
+		canUseFallback := statusCode != 401 && statusCode != 403 && statusCode != 404
+		if !canUseFallback {
+			utils.LogDebug(fmt.Sprintf("Received %v. Deleting (if exists) %v", statusCode, fallbackOpts.path))
+			os.Remove(fallbackOpts.path)
+			utils.LogDebug(fmt.Sprintf("Received %v. Deleting (if exists) %v", statusCode, fallbackOpts.legacyPath))
+			os.Remove(fallbackOpts.legacyPath)
+		}
+
```

Comment:
We should remove the metadata file while we're at it.

```suggestion
		if !canUseFallback {
			utils.LogDebug(fmt.Sprintf("Received %v. Deleting (if exists) %v", statusCode, fallbackOpts.path))
			os.Remove(fallbackOpts.path)
			utils.LogDebug(fmt.Sprintf("Received %v. Deleting (if exists) %v", statusCode, fallbackOpts.legacyPath))
			os.Remove(fallbackOpts.legacyPath)
			utils.LogDebug(fmt.Sprintf("Received %v. Deleting (if exists) %v", statusCode, metadataPath))
			os.Remove(metadataPath)
		}

```

Reply:
Done, nice catch 👍